packages/fhir-sanitize-param/index.js

```diff
@@ -72,142 +72,230 @@ function findMatchForConfig(name, params) {
-			// need to be validation done at the write level to make sure the
-			// resource is itself valid
-			// This will throw if value is not correct, no need to invariant
-			result = JSON.parse(value);
+		case 2:
+			system = chunks[0];
+			code = chunks[1];
 			break;
-		default:
-			// Force an invariant here
-			invariant(false, unsupportedError(type, name));
 	}
+	system = validator.stripLow(xss(sanitize(system)));
+	code = validator.stripLow(xss(sanitize(code)));
 
-	return result;
+	return {
+		code: code,
+		system: system
+	};
 }
 
 /**
- * @function parseArguments
- * @description Parse only arguments needed for this type of request
- * @param {Express.req} req - Request from an express server
- * @return {Object} - Arguments object
+ * @function sanitizeQuantity
+ * @description Sanitize quantity values
+ * @param name
+ * @param inputValue
+ * @param type
+ * @returns {*}
  */
-function parseArguments(req) {
-	let args = {};
-	// Merge query from get only
-	if (req.method === 'GET') {
-		args = Object.assign(args, req.query);
-	}
-	// Merge body from put and post only
-	if (/PUT|POST/.test(req.method)) {
-		args = Object.assign(args, req.body);
+function sanitizeQuantity(name, inputValue, type = 'quantity') {
+	let [number, token] = inputValue.split(/\|(.+)/);
+	number = sanitizeNumber(name, number, 'quantity.number');
+	let system = '';
+	let code = '';
+	if (token) {
+		({ system, code } = sanitizeToken(name, token, 'quantity.token'));
 	}
-	// Merge params for all requests
-	return Object.assign(args, req.params);
+	invariant(number, mismatchError(type, name));
+
+	return {
+		number: number,
+		system: system,
+		code: code,
+	};
+}
+
+/**
+ * @function coerceValue
+ * @description Given a value and a type, attempt to coerce the value to the correct type
+ * @param config
+ * @param requestValue
+ * @param isMissingQuery
+ * @returns {*}
+ */
+function coerceValue(config, requestValue, isMissingQuery) {
+	let { name, type } = config;
+	requestValue = Array.isArray(requestValue) ? requestValue : [requestValue];
+	const sanitizedResult = requestValue.map(values => {
+		invariant(typeof values === 'string', mismatchError(type, name));
+		return values.split(',').map(value => {
+			let result;
+
+			if (isMissingQuery) {
+				type = 'missing_query';
+			}
+			// Check all expected types
+			switch (type) {
+				case 'number':
+					result = sanitizeNumber(name, value);
+					break;
+				case 'date':
+					result = sanitizeDate(name, value);
+					break;
+				case 'uri':
+				case 'reference':
+				case 'string':
+					result = sanitizeString(name, value, type);
+					break;
+				case 'token':
+					result = sanitizeToken(name, value);
+					break;
+				case 'quantity':
+					result = sanitizeQuantity(name, value);
+					break;
+				case 'boolean':
+				case 'missing_query':
+					result = sanitizeBoolean(name, value);
+					break;
+				default:
+					// Force an invariant here
+					invariant(false, unsupportedError(type, name));
```

Comment:
I'm not sure if we should default to an error since params could be extensible and perhaps types as well...  Could we instead; pass this in as a 'raw' field that we couldn't coerce?

Reply:
We could! This may just be solved like the issue of arguments unspecified in the configs by just having core pass the whole request through

Reply:
Yup, I am gonna create a new PR here soon for a very generic response utils so I can generate controllers which will pass the sanitized args through and a context object containing the original request.